x/incentives/keeper/distribute.go

```diff
@@ -80,12 +88,12 @@ func (k Keeper) GetLocksToDistribution(ctx sdk.Context, distrTo lockuptypes.Quer
 	return []lockuptypes.PeriodLock{}
 }
 
-// getLocksToDistributionWithMaxDuration get locks that are associated to a condition
-// and if its by duration, then use the min Duration.
+// getLocksToDistributionWithMaxDuration returns locks that match the provided lockuptypes QueryCondition,
+// are greater than the provided minDuration, AND have yet to be distributed to.
 func (k Keeper) getLocksToDistributionWithMaxDuration(ctx sdk.Context, distrTo lockuptypes.QueryCondition, minDuration time.Duration) []lockuptypes.PeriodLock {
 	switch distrTo.LockQueryType {
 	case lockuptypes.ByDuration:
-		// TODO: FIX ME!!!!
+		// TODO: Create issue for this
 		// Confusingly, synthetic lockups aren't indexed by denom as expected.
 		// Thus you have to do this as a hack
 		denom := lockuptypes.NativeDenom(distrTo.Denom)
```

Comment:
Do we want to create an issue for this, or is this hack acceptable?

Reply:
I think we should keep track of such things in issues instead of code comments. Even if we don't plan to be immediately pursuing them

Reply:
https://github.com/osmosis-labs/osmosis/issues/2006